engineapi/backing_image_manager.go

```diff
@@ -69,40 +67,47 @@ func (c *BackingImageManagerClient) parseBackingImageFileInfo(bi *bimapi.Backing
 	}
 }
 
-func (c *BackingImageManagerClient) Fetch(name, uuid, fileName, checksum string, size int64) (*longhorn.BackingImageFileInfo, error) {
+func (c *BackingImageManagerClient) Fetch(name, uuid, checksum, dataSourceAddress string, size int64) (*longhorn.BackingImageFileInfo, error) {
 	if err := CheckBackingImageManagerCompatibilty(c.apiMinVersion, c.apiVersion); err != nil {
 		return nil, err
 	}
-	resp, err := c.grpcClient.Fetch(name, uuid, fileName, checksum, size)
+	resp, err := c.grpcClient.Fetch(name, uuid, checksum, dataSourceAddress, size)
 	if err != nil {
 		return nil, err
 	}
 	return c.parseBackingImageFileInfo(resp), nil
 }
 
-func (c *BackingImageManagerClient) Sync(name, uuid, checksum, fromHost, toHost string, size int64) (*longhorn.BackingImageFileInfo, error) {
+func (c *BackingImageManagerClient) Sync(name, uuid, checksum, fromHost string, size int64) (*longhorn.BackingImageFileInfo, error) {
 	if err := CheckBackingImageManagerCompatibilty(c.apiMinVersion, c.apiVersion); err != nil {
 		return nil, err
 	}
-	resp, err := c.grpcClient.Sync(name, uuid, checksum, fromHost, toHost, size)
+	resp, err := c.grpcClient.Sync(name, uuid, checksum, fmt.Sprintf("%s:%d", fromHost, BackingImageManagerDefaultPort), size)
 	if err != nil {
 		return nil, err
 	}
 	return c.parseBackingImageFileInfo(resp), nil
 }
 
-func (c *BackingImageManagerClient) Delete(name string) error {
+func (c *BackingImageManagerClient) PrepareDownload(name, uuid string) (string, string, error) {
+	if err := CheckBackingImageManagerCompatibilty(c.apiMinVersion, c.apiVersion); err != nil {
+		return "", "", err
+	}
+	return c.grpcClient.PrepareDownload(name, uuid)
+}
+
+func (c *BackingImageManagerClient) Delete(name, uuid string) error {
```

Comment:
I am curious why we need to use two unique identifies to do the operation? Why not just `Delete(uuid string)`? the same for Get.